### PR TITLE
[MI-2124] Added testcases for DefaultDate response 

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**

### CODE REVIEW

File: server/plugin/api_test.go [hunk 1]
```diff
@@ -716,3 +716,328 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
...
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s", PathDateTimeSelection),
+				Body: model.SubmitDialogRequest{
+					CallbackId: "mockPostID__Date",
+					Submission: map[string]interface{}{
+						"date": "2022-09-23",
+					},
+					ChannelId: "mockChannelID",
+				},
+			},
```
Comment (manojmalik20) [inline server/plugin/api_test.go:764]: This is being repeated in each test case. Reduce this repitition.

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -716,3 +716,328 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
...
+			userID: "mock-userID",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			p := new(Plugin)
+
+			mockAPI := &plugintest.API{}
+
+			mockAPI.On("GetBundlePath").Return("mockString", nil)
+
```
Comment (manojmalik20) [inline server/plugin/api_test.go:915]:
```suggestion
```

File: server/plugin/api_test.go [hunk 3]
```diff
@@ -716,3 +716,328 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
...
+			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 7)...).Return("LogDebug error")
+
+			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 6)...).Return("LogError error")
+
+			mockAPI.On("UpdatePost", mock.AnythingOfType("*model.Post")).Return(nil, nil)
```
Comment (manojmalik20) [inline server/plugin/api_test.go:920]:
```suggestion
			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 7)...).Return("")
			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 6)...).Return("")
			mockAPI.On("UpdatePost", mock.AnythingOfType("*model.Post")).Return(nil, nil)
```

File: server/plugin/api_test.go [hunk 4]
```diff
@@ -716,3 +716,328 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
...
+
+			p.SetAPI(mockAPI)
+
```
Comment (manojmalik20) [inline server/plugin/api_test.go:923]: Remove all the empty lines where it doesn't seem useful.

File: server/plugin/utils_test.go [hunk 5]
```diff
@@ -26,3 +27,92 @@ func Test_LogAndSendErrorToUser(t *testing.T) {
...
+		},
+		{
+			description: "Time is correct",
+			time:        "12:32",
+			expected:    "",
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			p := Plugin{}
+
```
Comment (manojmalik20) [inline server/plugin/utils_test.go:112]:
```suggestion
```
Same here, remove empty lines where not needed.

File: server/plugin/api_test.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/plugin/api_test.go:759]: The date/time is changing in most cases. In case you missed that!

Comment (manojmalik20) [reply server/plugin/api_test.go:759]: Still, there's a lot of repitition. I can see same body in 3-4 test cases. Even if the date/time is changing, the rest of the body is same. You can reduce this repetition by using functions.

Comment (raghavaggarwal2308) [reply server/plugin/api_test.go:759]: Done

File: server/plugin/api_test.go [hunk 6]
```diff
@@ -712,6 +712,17 @@ func TestPlugin_handlePickerSelection(t *testing.T) {
 	}
 }
 
+func getHandleDateTimeSelectionRequestBody(date, time, dialogType string) model.SubmitDialogRequest {
```
Comment (manojmalik20) [inline server/plugin/api_test.go:715]:
```suggestion
func getHandleDateTimeSelectionRequestBody(date, time, dialogType string) *model.SubmitDialogRequest {
```
Shouldn't it be like this?

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**